0_fund_analysis_functions_v2.R

```diff
@@ -0,0 +1,1210 @@
+      .data$isin
+    ) %>% 
+    summarise(market_value = sum(.data[[market_value_from]], ...))
+  # join with sensentive sector data 
+  .data <- .data %>% 
+    left_join(
+      sensentive_exposures, 
+      by = "isin"
+    ) %>% # and clean na.names 
+    mutate(sensitive_sector = if_else(is.na(.data$sensitive_sector), "other", .data$sensitive_sector))
+  # scaling things to avoid increasing the portfolio's market value size 
+  .data %>% 
+    group_by(
+      !!!rlang::syms(id_cols),
+      .data$isin
+    ) %>% 
+    mutate(adjusted_market_value = .data$market_value / n()) %>% 
+    ungroup()
+}
+
+load_esg_raw_data <- function() {
+  # define path
+  folder_path <- path("data", "iss_esg_data", "company_exposures")
+  # list files 
+  files <- list.files(folder_path)
+  # check file count 
+  stopifnot(n_distinct(files) == 11)
+  # load and map files 
+  data <- map_df(
+    files, 
+    function(x) {
+      readxl::read_xlsx(path(folder_path, x)) %>% 
+        mutate(source = x)
+    }
+  )
+  # clean input data 
+  data <- data %>% 
+    clean_names(case = "snake") %>% 
+    filter(str_detect(isin, "[[:upper:]]{2}[[:alnum:]]{10}"))
+  # extract sector 
+  data <- data %>% 
+    mutate(
+      sensitive_sector = str_remove(source, "2DII - "),
+      sensitive_sector = str_remove(sensitive_sector, "[[:punct:]][[:digit:]]{8}[[:punct:]]xlsx$"),
+      sensitive_sector = snakecase::to_any_case(sensitive_sector, case = "snake"),
+      sensitive_sector = str_remove(sensitive_sector, "_issuers$")
+    )
+  # add current data 
+  data <- data %>% 
+    mutate(last_updated = Sys.Date())
+  # make sure no files are dropped
+  stopifnot(n_distinct(data$source) == 11)
+  # save output
+  data %>% 
+    write_rds(path("data", "sensentive_sector_exposure", ext = "rds"))
+  # return data
+  return(data)
+}
+
+check_group_weight_parameters <- function(
+  data,
+  id_cols,
+  weights_from,
+  values_from,
+  values_to = NULL
+) {
+  # check parameters 
+  stopifnot(is.character(id_cols))
+  stopifnot(is.character(weights_from))
+  stopifnot(is.character(values_from))
+  stopifnot(is.character(values_to) | is.null(values_to))
+  # check class 
+  data %>% 
+    assertr::verify(class(.data[[weights_from]]) == "numeric") %>% 
+    assertr::verify(class(.data[[values_from]]) == "numeric")
+}
+
+
+summarise_group_weight <- function(
+  .data, 
+  id_cols,
+  weights_from,
+  values_from,
+  values_to = NULL, 
+  ...
+) {
+  # check parameters 
+  .data <- .data %>% 
+    check_group_weight_parameters(
+      id_cols,
+      weights_from,
+      values_from,
+      values_to
+    )
+  # weight value 
+  .data <- .data %>% 
+    dplyr::group_by(!!!rlang::syms(id_cols)) %>% 
+    dplyr::summarise(weighted_value := stats::weighted.mean(.data[[values_from]], .data[[weights_from]])) %>% 
+    dplyr::ungroup()
+  # rename output
+  if (!is.null(values_to)) {
+    .data %>% 
+      plyr::rename(c("weighted_value" = values_to))
+  } else if (is.null(values_to)) {
```

Comment:
In this case it seems enough to use `} else {`  instead of `} else if (condition) {`